app/services/badges/award_contributor_from_github.rb

```diff
@@ -9,19 +14,42 @@ class AwardContributorFromGithub
       "forem/DEV-ios",
     ].freeze
 
-    def self.call(since = 1.day.ago, msg = "Thank you so much for your contributions!")
-      return unless (badge_id = Badge.id_for_slug(BADGE_SLUG))
+    def self.call(msg = "Thank you so much for your contributions!")
+      new(msg).call
+    end
 
-      REPOSITORIES.each do |repo|
-        commits = Github::OauthClient.new.commits(repo, since: since.utc.iso8601)
+    def initialize(msg)
+      @msg = msg
+    end
 
-        authors_uids = commits.map { |commit| commit.author.id }
-        Identity.github.where(uid: authors_uids).find_each do |i|
-          BadgeAchievement
-            .where(user_id: i.user_id, badge_id: badge_id)
-            .first_or_create(rewarding_context_message_markdown: msg)
+    def call
+      REPOSITORIES.each do |repo|
+        commits = Github::OauthClient.new.commits(repo)
+        commit_authors_with_count = commits.each_with_object(Hash.new(0)) do |commit, hash|
+          hash[commit.author.id.to_s] += 1 unless commit.author.nil?
+        end
+        Identity.github.where(uid: commit_authors_with_count.keys).find_each do |identity|
+          create_badges_for_user(identity.user_id, commit_authors_with_count[identity.uid])
         end
       end
     end
+
+    private
+
+    attr_reader :msg
+
+    def create_badges_for_user(user_id, commits_count)
+      BADGE_SLUGS.each do |slug, milestone|
+        create_badge_achievement(user_id, slug, milestone, commits_count)
+      end
+    end
+
+    def create_badge_achievement(user_id, slug, milestone, commits_count)
+      return unless (badge_id = Badge.id_for_slug(slug))
+      return if commits_count < milestone
+
+      user_id_from_badges = BadgeAchievement.where(badge_id: badge_id).pluck(:user_id)
```

Comment:
@djuber Love the idea, my only concern here is what happens if there are many users who have reached a bronze level. Lets assume there are 100 users that have bronze(4 commits). So this would mean 100 queries to the database. Sounds reasonable?
Thats the only reason I went for memoize